arctic/store/bson_store.py

```diff
@@ -86,64 +86,64 @@ def insert_many(self, documents, **kwargs):
         self._arctic_lib.check_quota()
         return self._collection.insert_many(documents, **kwargs)
 
-    def delete_one(self, filter, **kwargs):
+    def delete_one(self, query_filter, **kwargs):
```

Comment:
I think we should keep this as `filter` as this matches the upstream pymongo API, and changing the keyword here is a minor API change.